syntax = "proto3";
package events;
import "google/protobuf/struct.proto";
import "google/protobuf/timestamp.proto";

service DatalakeManagerService {
    rpc InsertEventData (InsertEventRequest) returns (InsertEventResponse);
}

enum ValueType {
    VALUE_TYPE_UNSPECIFIED = 0;
    VALUE_TYPE_INT = 1;
    VALUE_TYPE_STRING = 2;
    VALUE_TYPE_BOOL = 3;
    VALUE_TYPE_LIST = 4;
}

message InsertEventRequest {
    string event_name = 1;
    string key = 2;
    google.protobuf.Timestamp date = 3;
    string project = 4;  // UUID as string
    string contact_urn = 5;  // optional, can be empty string or None
    ValueType value_type = 6;
    ValueData value = 7;
    google.protobuf.Struct metadata = 8;
}

message ValueData {
    oneof value {
        int64 int_value = 1;
        string string_value = 2;
        bool bool_value = 3;
        StringList list_value = 4;
    }
}

message StringList {
    repeated string values = 1;
}

message InsertEventResponse {
    string status = 1;
}
